pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -132,6 +147,153 @@ func storageMatches(set *apps.StatefulSet, pod *v1.Pod) bool {
 	return true
 }
 
+// claimOwnerMatchesSetAndPod returns false if the ownerRefs of the claim are not set consistently with the
+// PVC deletion policy for the StatefulSet.
+func claimOwnerMatchesSetAndPod(claim *v1.PersistentVolumeClaim, set *apps.StatefulSet, pod *v1.Pod) bool {
+	switch set.Spec.PersistentVolumeClaimDeletePolicy {
+	case apps.Retain:
+		if hasOwnerRef(claim, set) ||
+			hasOwnerRef(claim, pod) {
+			return false
+		}
+	case apps.DeleteOnStatefulSetDeletionOnly:
+		if !hasOwnerRef(claim, set) ||
+			hasOwnerRef(claim, pod) {
+			return false
+		}
+	case apps.DeleteOnScaledownOnly:
+		if hasOwnerRef(claim, set) {
+			return false
+		}
+		podScaledDown := getOrdinal(pod) >= int(*set.Spec.Replicas)
```

Comment:
With [parallel pod management](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#parallel-pod-management) can we still make the assumption that the highest ordinals are scaled down first?

Reply:
My understanding with parallel pod management is that Replicas will still be set to the final number of replicas. We just don't know which of the condemned pods will get deleted first. 

In other words, with both parallel pod management as well as conventional, all the condemned pods will have ordinal > Replicas, and we will set owner refs on all of those PVCs at the same time. We don't need to do anything different w/rt PVC deletion AFAICT.

Reply:
this is correct, all condemned pods will have ordinal > Replicas